source/extensions/transport_sockets/tls/ssl_handshaker.h

```diff
@@ -36,63 +36,27 @@ class SslExtendedSocketInfoImpl : public Envoy::Ssl::SslExtendedSocketInfo {
       Envoy::Ssl::ClientValidationStatus::NotValidated};
 };
 
-class SslHandshakerImpl : public Ssl::ConnectionInfo,
+class SslHandshakerImpl : public ConnectionInfoImpl,
                           public Ssl::Handshaker,
                           protected Logger::Loggable<Logger::Id::connection> {
 public:
   SslHandshakerImpl(bssl::UniquePtr<SSL> ssl, int ssl_extended_socket_info_index,
                     Ssl::HandshakeCallbacks* handshake_callbacks);
 
-  // Ssl::ConnectionInfo
-  bool peerCertificatePresented() const override;
+  // ConnectionInfoImpl
   bool peerCertificateValidated() const override;
```

Comment:
I would expect this `peerCertificateValidated` to be implemented in `ConnectionInfoImpl` because of its name but looks like it doesn't and it is/will be implemented by SslHandshakerImpl and QUIC stuff. That seems not weird to me. Maybe we should move `peerCertificateValidated` from Ssl::ConnectionInfo interface or can it be implemented inside of `ConnectionInfoImpl`?

Reply:
peerCertificateValidated() will be implemented differently in QUIC. So I didn't move it over.  That being said, probably the class name should be ConnectionInfoImplBase to make it clear that it's a virtual class.

Reply:
> the class name should be ConnectionInfoImplBase to make it clear that it's a virtual class.

sounds good!

Reply:
done